resolvers/npm_cache/lib.rs

```diff
@@ -137,40 +142,57 @@ impl NpmCacheSetting {
   }
 }
 
-/// Stores a single copy of npm packages in a cache.
-#[derive(Debug)]
-pub struct NpmCache<
-  TSys: FsCreateDirAll
+pub trait NpmCacheSys:
```

Comment:
Wonder if we should add a macro to `sys_traits` to generate these aliases. Something like

```
macro_rules! sys_traits_alias {
    ($name : ident : $($t: path),* $(,)?) => {
      pub trait $name: $($t +)* { }
      
      impl<T: 'static> $name for T where
        T: $($t +)+
      {
      }
    };
}
```
used like
`sys_trait_alias!(NpmCacheSys: FsCanonicalize, FsCreateDirAll, ...)`

Reply:
Yeah, it would be useful. I opened https://github.com/dsherret/sys_traits/issues/43